File: yh_tutorial/yh_tutorial_7/src/yh_server_7.cpp
#include "ros/ros.h"
#include "yh_tutorial_7/yh_srv_7.h"

bool srvCallback(yh_tutorial_7::yh_srv_7::Request &req, yh_tutorial_7::yh_srv_7::Response &res)
{
    if (req.a > req.b)
    {
        res.result = req.a - req.b;
    }
    else
    {
        res.result = req.b - req.a;
    }

    return true;
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "yh_server_7");
    ros::NodeHandle nh;

    ros::ServiceServer service_server = nh.advertiseService("yh_service_7", srvCallback);

    ROS_INFO("Service server ready.");

    ros::spin();

    return 0;
}

File: yh_tutorial/yh_tutorial_7/scripts/yh_server_7.py
#!/usr/bin/python
# -*- coding: utf8 -*-

import rospy
from yh_tutorial_7.srv import yh_srv_7, yh_srv_7Response


def srvCallback(req):
    if req.a > req.b:
        return yh_srv_7Response(req.a - req.b)
    else:
        return yh_srv_7Response(req.b - req.a)


def yh_server():
    rospy.init_node("yh_server_7")
    service_server = rospy.Service("yh_service_7", yh_srv_7, srvCallback)
    rospy.spin()


if __name__ == "__main__":
    yh_server()
